### ISSUE TITLE
"async & performance": Chapter 2 - Trying to save callbacks

### ISSUE THREAD

**Issue by utullo** [getify/You-Dont-Know-JS#256]
First of all, thanks for your GREAT job.
I really appreciate your work: your pubblications REALLY helped me in understanding JS topics you covered.
So thanks.  A lot.
I just read Chapter 2 of "async & performance" and I have a question about the section "Trying to save callbacks".
In the last example of <code>asyncify()</code> I can't understand the meaning of the returning function:

<code>
        // firing too quickly, before `intv` timer has fired to<br/>
</code>
<code>
        // indicate async turn has passed?
</code>
<code>
        if (intv) {
</code>
<code>
&nbsp;&nbsp;            fn = orig_fn.bind.apply(
</code>
<code>
&nbsp;&nbsp;&nbsp;&nbsp;               orig_fn,
</code>
<code>
 &nbsp;&nbsp;&nbsp;&nbsp;               // add the wrapper's `this` to the `bind(..)`
</code>
<code>
 &nbsp;&nbsp;&nbsp;&nbsp;               // call parameters, as well as currying any
</code>
<code>
&nbsp;&nbsp;&nbsp;&nbsp;                // passed in parameters
</code>
<code>
&nbsp;&nbsp;&nbsp;&nbsp;                [this].concat( [].slice.call( arguments ) )
</code>
<code>
&nbsp;&nbsp;            );
</code>
<code>
        }
</code>
<code>
        // already async
</code>
<code>
        else {
</code>
<code>
&nbsp;&nbsp;            // invoke original function
</code>
<code>
 &nbsp;&nbsp;           orig_fn.apply( this, arguments );
</code>
<code>
        }
</code>

IE: what's the meaning of <code>fn = orig_fn.bind.apply(..)</code> block ?  Why restoring <code>fn</code> function has to be done this way ?  What's <code>this</code> value ?
Basically that's because I am relatively new to Javascript, but I think that a more detailed explanation of the snippet above or a reference to same external resource could be useful for not-so-advanced JS programmers like me.

Thanks
